addons/link_tracker/tests/test_merge_utm_campaign.py

```diff
@@ -0,0 +1,91 @@
+    def setUp(self):
+        result = super(TestMergeUtmCampaign, self).setUp()
+        self.UtmCampaign = self.env['utm.campaign']
+        self.LinkTracker = self.env['link.tracker']
+        self.LinkTrackerClick = self.env['link.tracker.click']
+        self.utm_source = self.env['utm.source'].create({
+            'name': 'utm source'
+        })
+        self.utm_medium = self.env['utm.medium'].create({
+            'name': 'utm medium'
+        })
+        self.utm_campaign_1 = self.UtmCampaign.create({
+            'name': 'utm campaign 1'
+        })
+        self.utm_campaign_2 = self.UtmCampaign.create({
+            'name': 'utm campaign 2'
+        })
+        self.utm_campaign_3 = self.UtmCampaign.create({
+            'name': 'utm campaign 3'
+        })
+        self.link_tracker_1 = self.LinkTracker.create({
+            'url': 'www.dodd.com',
+            'campaign_id': self.utm_campaign_1.id,
+            'medium_id': self.utm_medium.id,
+            'source_id': self.utm_source.id
+        })
+        self.link_tracker_2 = self.LinkTracker.create({
+            'url': 'www.dodd.com',
+            'campaign_id': self.utm_campaign_2.id,
+            'medium_id': self.utm_medium.id,
+            'source_id': self.utm_source.id
+        })
+        self.link_tracker_3 = self.LinkTracker.create({
+            'url': 'www.odoo.com',
+            'campaign_id': self.utm_campaign_3.id,
+            'medium_id': self.utm_medium.id,
+            'source_id': self.utm_source.id
+        })
+        self.link_tracker_click_1 = self.LinkTrackerClick.create({
+            'link_id': self.link_tracker_1.id
+        })
+        self.link_tracker_click_2 = self.LinkTrackerClick.create({
+            'link_id': self.link_tracker_2.id
+        })
+        self.link_tracker_click_3 = self.LinkTrackerClick.create({
+            'link_id': self.link_tracker_3.id
+        })
+        return result
+
+    def test_merge_unique(self):
+        link_codes_1 = self.link_tracker_1.link_code_ids
+        link_clicks_1 = self.link_tracker_1.link_click_ids
+        link_codes_2 = self.link_tracker_2.link_code_ids
+        link_clicks_2 = self.link_tracker_2.link_click_ids
+
+        (self.utm_campaign_1 + self.utm_campaign_2)._merge_utm_campaigns(self.utm_campaign_1)
+
+        link_tracker = self.env['link.tracker'].search([('campaign_id', '=', self.utm_campaign_1.id)])
+
+        self.assertEqual(len(link_tracker.ids), 1, 
+            "After merging utm.campaings, duplicate link.trackers should be unlinked.")
+        self.assertEqual(link_tracker.link_code_ids, link_codes_1 + link_codes_2, 
+            "After merging utm.campaings, link.tracker.code of duplicate link.tracker should be redirected to the remaining one.")
+        self.assertEqual(link_tracker.link_click_ids, link_clicks_1 + link_clicks_2, 
+            "After merging utm.campaings, link.tracker.click of duplicate link.tracker should be redirected to the remaining one.")
```

Comment:
You could also check that:

- The link_tracker_2.campaign_id is equal to utm_campaign_1
- The utm_campaign_2 is archived
- The utm_campaign_2.referenced_campaign_id is equal to utm_campaign_1